MMCore/BufferManager.cpp

```diff
@@ -0,0 +1,384 @@
+   return useV2_.load();
+}
+
+int BufferManager::ReleaseReadAccess(const void* ptr) {
+    if (useV2_.load() && ptr) {
+        return v2Buffer_->ReleaseDataReadPointer(ptr);
+    }
+    return DEVICE_ERR;
+}
+
+unsigned BufferManager::GetDataSize(const void* ptr) const {
+   if (!useV2_.load()) 
+      return circBuffer_->GetImageSizeBytes();
+   else
+      return static_cast<long>(v2Buffer_->GetDatumSize(ptr));
+}
+
+int BufferManager::SetOverwriteData(bool overwrite) {
+    if (useV2_.load()) {
+        return v2Buffer_->SetOverwriteData(overwrite);
+    } else {
+        return circBuffer_->SetOverwriteData(overwrite);
+    }
+}
+
+int BufferManager::AcquireWriteSlot(const char* deviceLabel, size_t dataSize, size_t additionalMetadataSize,
+    void** dataPointer, void** additionalMetadataPointer, Metadata* pInitialMetadata) {
+   if (!useV2_.load()) {
+      // Not supported for circular buffer
+      return DEVICE_ERR;
+   }
+
+   // Initialize metadata with either provided metadata or create empty
+   Metadata md = (pInitialMetadata != nullptr) ? *pInitialMetadata : Metadata();
+   
+   std::string serializedMetadata = md.Serialize();
+   int ret = v2Buffer_->AcquireWriteSlot(dataSize, additionalMetadataSize,
+      dataPointer, additionalMetadataPointer, serializedMetadata, deviceLabel);
+   return ret;
+}
+
+int BufferManager::FinalizeWriteSlot(const void* imageDataPointer, size_t actualMetadataBytes) {
+    if (!useV2_.load()) {
+        // Not supported for circular buffer
+        return DEVICE_ERR;
+    }
+    return v2Buffer_->FinalizeWriteSlot(imageDataPointer, actualMetadataBytes);
+}
+
+void BufferManager::ExtractMetadata(const void* dataPtr, Metadata& md) const {
+    if (!useV2_.load()) {
+        throw CMMError("ExtractMetadata is only supported with V2 buffer enabled");
+    }
+    
+    if (v2Buffer_ == nullptr) {
+        throw CMMError("V2 buffer is null");
+    }
+
+    int result = v2Buffer_->ExtractCorrespondingMetadata(dataPtr, md);
+    if (result != DEVICE_OK) {
+        throw CMMError("Failed to extract metadata");
+    }
+}
+
+const void* BufferManager::GetLastDataFromDevice(const std::string& deviceLabel) throw (CMMError) {
+    if (!useV2_.load()) {
+        throw CMMError("V2 buffer must be enabled for device-specific data access");
+    }
+    Metadata md;
+    return GetLastDataMDFromDevice(deviceLabel, md);
+}
+
+const void* BufferManager::GetLastDataMDFromDevice(const std::string& deviceLabel, Metadata& md) throw (CMMError) {
+    if (!useV2_.load()) {
+        throw CMMError("V2 buffer must be enabled for device-specific data access");
+    }
+    
+    const void* basePtr = v2Buffer_->PeekLastDataReadPointerFromDevice(deviceLabel, md);
+    if (basePtr == nullptr) {
+        throw CMMError("No data found for device: " + deviceLabel, MMERR_InvalidContents);
+    }
+    return basePtr;
+}
+
+bool BufferManager::IsPointerInV2Buffer(const void* ptr) const throw (CMMError) {
```

Comment:
I got a compile-time error here:

```
      mmCoreAndDevices/MMCore/BufferManager.cpp:357:21: error: exception specification in declaration does not match previous declaration
        357 | bool BufferManager::IsPointerInV2Buffer(const void* ptr) const throw (CMMError) {
            |                     ^
      mmCoreAndDevices/MMCore/BufferManager.h:267:9: note: previous declaration is here
        267 |    bool IsPointerInV2Buffer(const void* ptr) const;
            |         ^
      1 error generated.
```

Reply:
Hint: `throw (CMMError)` in MMCore is purely for SWIG's benefit, so it doesn't need to be added to internal functions.